Added the ability to add descriptions to build tasks.

TxN opened the pull request:
+ Added new fields to Build JSON's: "short_description", "long_description", "args_description"
+ Added "help list" and "help build_task" commands
+ Modified "Help" command text output, reformatted text messages of the builder.

Commit:
Builds desсription support.

Commit:
Some formatting tweaks and improvements, improved messages readability, added arguments descriptions for build tasks.

Commit:
Fixed arg count in error (no need to subtract 1 from args count)

Commit:
Couple tweaks of builder slack messages.

Commit:
Fixed bugs which prevented help for build tasks to work.

KonH commented on Server/Services/StatService.cs:
```diff
@@ -30,8 +30,8 @@ public class StatService : IService {
 			_server.OnInitBuild += OnInitBuild;
-			_server.AddCommand("stats", "show stats about builds", OnStatsRequested);
-			_server.AddCommand("history", "show history about builds", OnHistoryRequested);
+			_server.AddCommand("stats", "show builds stats (build cout, min,max,average build times)", OnStatsRequested);
```
Should be (build count, min, max, average build times)

KonH commented on Server/Views/BaseServerView.cs:
```diff
@@ -140,9 +172,32 @@ class ContextWrapper {
+				sb.Append("No arguments.");
+				return;
+			}
+			var outputDescriptions = build.ArgsDescription.Count == build.Args.Count;
```
Looks strange, cases, when description count is less can happen (e.g. new arg added, but without description, it such cases description for all args disappear).

KonH commented:
Looks nice! But we need to two fixes by review and please add this fields in README.md.

Commit:
Fixed typo in 'stats' command text output

Commit:
Updated README.md

Commit:
Fixed argument descriptions output.

(now it outputs args descriptions even if there are fewer descriptions than argument count)

TxN commented:
Ok!
I've fixed these issues and updated readme.
